micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java

```diff
@@ -41,14 +41,24 @@ private JvmMemory() {
     static boolean isConcurrentPhase(String cause, String name) {
         return "No GC".equals(cause)
                 || "Shenandoah Cycles".equals(name);
+                // Don't think this applies to J9 - maybe "J9MMCONSTANT_IMPLICIT_GC_COMPLETE_CONCURRENT" / concurrent collection must be completed"
     }
 
     static boolean isYoungGenPool(String name) {
-        return name != null && name.endsWith("Eden Space");
+        return name != null && (name.endsWith("Eden Space") 
+            || name.endsWith("nursery-allocate") 
+            || name.endsWith("-survivor") // "nursery-survivor" & "balanced-survivor"
+            || name.endsWith("-eden") // "balanced-eden"
+        );
     }
 
     static boolean isOldGenPool(String name) {
-        return name != null && (name.endsWith("Old Gen") || name.endsWith("Tenured Gen"));
+        return name != null && (name.endsWith("Old Gen")
+            || name.endsWith("Tenured Gen")
+            || name.endsWith("balanced-old")
+            || name.endsWith("balanced-reserved")
+            || name.contains("tenured") // "tenured", "tenured-SOA", "tenured-LOA"
```

Comment:
The `optavgpause` and `optthruput` collectors are non-generational, but we can't detect that from the memory pool name alone since it uses `tenured-SOA` and `tenured-LOA` the same as gencon does. For generational collectors, we calculate the allocated bytes from the young (eden/nursery) pool, but for non-generational collectors we can't do that. Maybe we can rework the current logic somehow without too much special casing.

Reply:
I added some less than ideal logic for detecting generational vs non in #2560. It works with the current implementation of all garbage collectors we claim to support even after this pull request, but it does seem fragile.